src/java/org/jivesoftware/LoginDialog.java

```diff
@@ -277,11 +239,22 @@ protected XMPPTCPConnectionConfiguration retrieveConnectionConfiguration() {
         }
 
         final XMPPTCPConnectionConfiguration.Builder builder = XMPPTCPConnectionConfiguration.builder()
-                .setUsernameAndPassword( "username", "password" )
+                .setUsernameAndPassword( loginUsername, loginPassword )
```

Comment:
Wow. I wonder why it is working for me in the first place (no, my username is not "username").